### DISCUSSION TITLE
Forcing structure offset detection

### DISCUSSION THREAD

**Discussion by baileyn** [Vector35/binaryninja-api#4626] [Q&A] [answered]
Hey all! I'm new to Binary Ninja and I'm trying to test it out a bit. If this is the wrong category, I apologize.

I'm reversing an application that happens to have a very large structure (at least 0x700000 bytes). This causes some craziness at times, but I can usually push my way through. I've ran in to this issue with BN and I was curious if there's a way for me to force it to be interpreted correctly.

I have the following line:
![image](https://github.com/Vector35/binaryninja-api/assets/6194343/ae87626d-d8af-4214-8247-e6fa270808b4)

Essentially, what's happening here (from my work in other programs) is the following:

```c
pObj->rootWidgetXs[pObject->rootWidgetCount] = pOther->x + paddingX;
```

The primary issue is that I'm not sure how to coerce this into seeing 0x20f514 as an offset into the Obj structure so I can retype it appropriately (int[100]). Is there a way to achieve this?

**Comment by plafosse** [comment]
What is your `pObj` structure defined as?

**Comment by baileyn** [comment]
It's currently just a big structure with several `byte gap[102392]` fields that are somewhat guessed. Unfortunately this program has a very big (over 0x700000 bytes) state structure that stores pretty much everything. 

I'm not entirely sure how to export the structure entirely, but here's the section that's being used in the above example:
![image](https://github.com/Vector35/binaryninja-api/assets/6194343/cb28ce6a-5624-472e-b5ac-068d63d789c3)

I suppose another question that may resolve this is how I could go about undefining the arrays that were auto generated. I've tried `u` and I've tried looking in the context menu for `undefine`, but I don't see anything that looks like it would do it.

**Comment by fuzyll** [comment]
I believe this is an existing issue: #4251. Does this sound the same to you? If so, this is the answer until that gets fixed. If not, we should maybe turn this discussion into a separate issue and look into it further.

**Comment by plafosse** [comment, marked as answer]
Andrew articulated the root cause of this but in the following issue: https://github.com/Vector35/binaryninja-api/issues/4749
